src/workloads/query/LookupColocatedData.yml

```diff
@@ -0,0 +1,368 @@
+                    as: "joined"
+                  }},
+                ]
+              cursor: {batchSize: {^Parameter: {Name: "BatchSize", Default: 101}}}
+
+Actors:
+- Name: CreateShardedCollections
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *NumPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        Operations:
+        - OperationMetricsName: EnableSharding
+          OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database
+        # This collection will have data distributed across 6 chunks.
+        - OperationMetricsName: ShardLocalCollection
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: test.ShardedLocal
+            key: {*LocalShardKey: hashed}
+            numInitialChunks: 6
+        # Here we use 1 chunk. This ensures that all the data for this coll is on one shard.
+        - OperationMetricsName: ShardLocalCollectionAllOnOneShard
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: test.ShardedLocalAllOnOneShard
+            key: {*LocalShardKey: hashed}
+            numInitialChunks: 1
+        # Disable the balancer so that it can't skew results while the $lookups are running.
+        - OperationMetricsName: DisableBalancer
+          OperationName: AdminCommand
+          OperationCommand:
+            balancerStop: 1
+
+# On the sharded collections, an index wil automatically be created to support the shard key. Here
+# we create an equivalent index on the unsharded collection.
+- Name: CreateIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *NumPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Operations:
+        - OperationName: RunCommand
+          OperationCommand:
+            createIndexes: *UnshardedLocal
+            indexes:
+            - key: {*LocalShardKey: 1}
+              name: localShardKey_1
+
+- ActorFromTemplate:
+    TemplateName: LoadDocuments
+    TemplateParameters:
+      OnlyActiveInPhase: 2
+      Coll: *UnshardedLocal
+
+- ActorFromTemplate:
+    TemplateName: LoadDocuments
+    TemplateParameters:
+      OnlyActiveInPhase: 3
+      Coll: *ShardedLocal
+
+- ActorFromTemplate:
+    TemplateName: LoadDocuments
+    TemplateParameters:
+      OnlyActiveInPhase: 4
+      Coll: *ShardedLocalAllOnOneShard
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [5]
+      NopInPhasesUpTo: *NumPhases
+      PhaseConfig:
+        Repeat: 1
+
+#
+# Unsharded
+#
+- ActorFromTemplate:
+    TemplateName: SelfLookupLocalFieldForeignField
+    TemplateParameters:
+      OnlyActiveInPhase: 6
+      MetricsName: UnshardedLocalFieldForeignField1k
+      Coll: *UnshardedLocal
+      BatchSize: *BatchSize1k
```

Comment:
I don't have any concrete data to support this, but I feel like most customer workloads are not setting `batchSize` explicitly. Was it different in the HELP ticket that these came from? I'm wondering if we should also have a case for the default/unspecified batchSize

Reply:
See my response above. I'm pretty sure we need to specify a batchSize for the actors, so we could use the default. It would be fine with me to add that case, but it was interesting to me to see the change in perf characteristics as the batchSize (aka amount of data processed) increased.